service/header/store.go

```diff
@@ -182,23 +191,25 @@ func (s *store) Append(ctx context.Context, headers ...*ExtendedHeader) error {
 	}
 
 	verified := make([]*ExtendedHeader, 0, lh)
-	for _, h := range headers {
-		if head.Height == h.Height {
-			continue
-		}
-
+	for i, h := range headers {
 		err = VerifyAdjacent(head, h)
 		if err != nil {
-			log.Errorw("invalid header", "current head", head.Hash(), "height",
-				head.Height, "attempted new header", h.Hash(), "height", h.Height, "err", err)
-			break // if some headers are cryptographically valid, why not include them? Exactly, so let's include
+			if i == 0 {
+				return err
+			}
+
+			var verErr *VerifyError
+			if errors.As(err, &verErr) {
+				log.Errorw("invalid header",
+					"current height", head.Height,
+					"height", head.Height,
+					"hash", h.Hash(),
+					"reason", err)
+				break
+			}
```

Comment:
why use errors.As instead of errors.Is here? you're not using `verErr`

Reply:
Oh, thank you for catching this. Should use verErr.Reason instead